### PR TITLE
Increase psalm level to 3 and corresponding changes

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: src/Collector/LoggerInterfaceProxy.php [hunk 1]
```diff
@@ -50,49 +50,58 @@ public function critical(string|Stringable $message, array $context = []): void
...
         $this->logger->log($level, $message, $context);
     }
+
+    /**
+     * @psalm-return array{file: string, line: int}
+     */
+    private function getCallStack(): array
+    {
+        /** @psalm-var array{file: string, line: int} */
+        return debug_backtrace()[0];
```
Comment (xepozz) [inline src/Collector/LoggerInterfaceProxy.php:105]: You should add +1 because stack has changed

File: src/Collector/Console/CommandCollector.php [hunk 2]
```diff
@@ -68,13 +68,14 @@ public function collect(ConsoleEvent|ConsoleErrorEvent|ConsoleTerminateEvent $ev
             return;
         }
 
+        $definition = $command?->getDefinition();
         $this->commands[$event::class] = [
-            'name' => $command->getName(),
+            'name' => $command?->getName() ?? $event->getInput()->getFirstArgument() ?? '',
```
Comment (xepozz) [inline src/Collector/Console/CommandCollector.php:73]: Are you sure that first argument is a command name?

**Review by xepozz — CHANGES_REQUESTED** [review]

File: src/Collector/LoggerInterfaceProxy.php (hunk 1 above)
Comment (vjik) [reply src/Collector/LoggerInterfaceProxy.php:105]: Why? It change current behaviour. 

In PR replaced `[$callstack] = debug_backtrace()` to `$callstack = debug_backtrace()[0]`. It is same, isn't it?

File: src/Collector/Console/CommandCollector.php (hunk 2 above)
Comment (vjik) [reply src/Collector/Console/CommandCollector.php:73]: I don't know. I spied it here:

image

File: src/Collector/LoggerInterfaceProxy.php (hunk 1 above)
Comment (xepozz) [reply src/Collector/LoggerInterfaceProxy.php:105]: Yes, but you have moved the function call into another function

Comment (vjik) [reply src/Collector/LoggerInterfaceProxy.php:105]: Oh. I understand. 
I will fix it and try create test.

Comment (vjik) [reply src/Collector/LoggerInterfaceProxy.php]: Fixed

**Review by xepozz — APPROVED** [review]

**Pull request merged**